src/Plugin.php

```diff
@@ -5,34 +5,227 @@
 
-            // If we're running a Laravel container, let's see if we need to register the IDE helper if it isn't
-            // already. If we don't do this, the plugin will crash out because the IDE helper doesn't have configs
-            // it bootstraps present in the app container.
+        $models_generator_command->run(
+            new \Symfony\Component\Console\Input\ArrayInput([
+                '--nowrite' => true
+            ]),
+            new \Symfony\Component\Console\Output\NullOutput()
+        );
 
-            if (!$app->getProvider($ide_helper_provider)) {
-                $app->register($ide_helper_provider);
-            }
+        $registration->addStubFile($cache_dir . 'models.php');
+
+        self::$model_classes = $models_generator_command->getModels();
+    }
+
+    /**
+     * @param \Illuminate\Foundation\Application|\Laravel\Lumen\Application $app
+     * @param FakeFilesystem $fake_filesystem
+     * @return Factory
+     */
+    private function getViewFactory(
+        \Illuminate\Container\Container $app,
+        FakeFilesystem $fake_filesystem
+    ) : Factory {
+        $service_helper_reflection = new \ReflectionClass(\Barryvdh\LaravelIdeHelper\IdeHelperServiceProvider::class);
+
+        $file_path = $service_helper_reflection->getFileName();
+
+        if (!$file_path) {
+            throw new \UnexpectedValueException('Service helper should have a file path');
         }
+
+        $resolver = new EngineResolver();
+        $resolver->register('php', function () : PhpEngine {
+            return new PhpEngine();
+        });
+        $finder = new FileViewFinder($fake_filesystem, [dirname($file_path) . '/../resources/views']);
+        $factory = new Factory($resolver, $finder, new \Illuminate\Events\Dispatcher());
+        $factory->addExtension('php', 'php');
+        return $factory;
+    }
+
+    /**
+     * @param \Illuminate\Foundation\Application $app
+     */
+    protected function getEnvironmentSetUp($app): void
+    {
+        $app['config']->set('app.key', 'AckfSECXIvnK5r28GVIWUAxmbBSjTsmF');
+    }
+
+    private function addOurStubs(RegistrationInterface $registration): void
+    {
+        foreach (glob(__DIR__ . '/Stubs/*.stubphp') as $stubFilePath) {
+            $registration->addStubFile($stubFilePath);
+        }
+    }
+
+    /**
+     * @return \Illuminate\Foundation\Application|\Laravel\Lumen\Application
+     */
+    private function bootApp()
+    {
+        if (file_exists($applicationPath = __DIR__.'/../../../../bootstrap/app.php')) { // Applications
+            $app = require $applicationPath;
+        } elseif (file_exists($applicationPath = getcwd().'/bootstrap/app.php')) { // Local Dev
+            $app = require $applicationPath;
+        } else { // Packages
+            $app = (new static)->createApplication();
+        }
+
+        if ($app instanceof \Illuminate\Contracts\Foundation\Application) {
```

Comment:
I find having these two conditional statements here much more readable than splitting the logic via inheritance